crates/re_renderer/src/resource_managers/texture_manager.rs

```diff
@@ -95,6 +95,32 @@ impl<'a> Texture2DCreationDesc<'a> {
     }
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum TextureCreationError {
+    #[error("Texture with debug label {0:?} has zero width or height!")]
+    ZeroSize(DebugLabel),
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum TextureManager2DError<DataCreationError> {
+    /// Something went wrong when creating the GPU texture.
+    #[error(transparent)]
+    TextureCreation(TextureCreationError),
+
+    /// Something went wrong in a user-callback.
+    #[error(transparent)]
+    DataCreation(#[from] DataCreationError),
```

Comment:
after writing this I realized this would probably lead to less code, and less complications:

```suggestion
    #[error(transparent)]
    TextureCreation(#[from] TextureCreationError),

    /// Something went wrong in a user-callback.
    #[error(transparent)]
    DataCreation(DataCreationError),
```